### PR TITLE
Add local d1 migrations

### PR DESCRIPTION
**Pull request by rozenmd**

### CODE REVIEW

File: packages/wrangler/src/d1/migrations/apply.tsx [hunk 1]
```diff
@@ -21,7 +21,17 @@ import type { BaseSqlExecuteArgs } from "../execute";
 import type { Argv } from "yargs";
 
 export function ApplyOptions(yargs: Argv): Argv<BaseSqlExecuteArgs> {
-	return Database(yargs);
+	return Database(yargs)
+		.option("local", {
+			describe:
+				"Execute commands/files against a local DB for use with wrangler dev",
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/apply.tsx:27]:
```suggestion
				"Execute commands/files against a local DB for use with wrangler dev --local",
```

File: packages/wrangler/src/d1/migrations/list.tsx [hunk 2]
```diff
@@ -16,7 +16,17 @@ import type { BaseSqlExecuteArgs } from "../execute";
...
+		.option("local", {
+			describe:
+				"Execute commands/files against a local DB for use with wrangler dev",
+			type: "boolean",
+		})
+		.option("persist-to", {
+			describe: "Specify directory to use for local persistence (for --local)",
+			type: "string",
+			requiresArg: true,
+		});
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/list.tsx:29]: Is there any way these options can be unified? Specifying them twice seems a bit redundant

File: packages/wrangler/src/d1/migrations/apply.tsx [hunk 3]
```diff
@@ -21,7 +21,17 @@ import type { BaseSqlExecuteArgs } from "../execute";
...
+			type: "boolean",
+		})
+		.option("persist-to", {
+			describe: "Specify directory to use for local persistence (for --local)",
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/apply.tsx:31]: Can we use something like https://yargs.js.org/docs/#implies to enforce the "for `--local`" relationship?

**Review by penalosa — CHANGES_REQUESTED** [review]

**Comment by rozenmd** [comment]
Unfortunately this wasn't as simple as passing the flags through, need to debug why it can't find the local sqlite file

File: packages/wrangler/src/d1/migrations/list.tsx (hunk 2 above)
Comment (rozenmd) [reply packages/wrangler/src/d1/migrations/list.tsx:29]: I could probably define them centrally (like we do with `Database`) and export

File: packages/wrangler/src/d1/migrations/apply.tsx [hunk 4]
```diff
@@ -88,8 +99,7 @@ export const ApplyHandler = withConfig<BaseSqlExecuteArgs>(
 			return;
 		}
 
-		const isInteractive = process.stdout.isTTY;
-		if (isInteractive) {
+		if (isInteractive()) {
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/apply.tsx:102]: In most other places we use the check `isInteractive() && !CI.isCI()`. I'm working on adding that check into the `confirm` function, so this isn't a blocker for this PR, but just wanted to note it.

File: packages/wrangler/src/d1/migrations/apply.tsx [hunk 5]
```diff
@@ -121,7 +131,7 @@ export const ApplyHandler = withConfig<BaseSqlExecuteArgs>(
 					local,
 					config,
 					database,
-					undefined,
+					isInteractive(),
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/apply.tsx:134]: This, however, should definitely use the check `isInteractive() && !CI.isCI()`, I think

File: packages/wrangler/src/d1/migrations/helpers.ts [hunk 6]
```diff
@@ -79,7 +80,7 @@ const listAppliedMigrations = async (
 		local,
 		config,
 		name,
-		undefined,
+		isInteractive(),
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/helpers.ts:83]: As above

File: packages/wrangler/src/d1/migrations/helpers.ts [hunk 7]
```diff
@@ -130,7 +131,7 @@ export const initMigrationsTable = async (
...
```
Comment (penalosa) [inline packages/wrangler/src/d1/migrations/helpers.ts:134]: As above

File: packages/wrangler/src/d1/migrations/list.tsx (hunk 2 above)
Comment (penalosa) [reply packages/wrangler/src/d1/migrations/list.tsx:29]: Resolving pending followup

File: packages/wrangler/src/d1/migrations/apply.tsx (hunk 3 above)
Comment (penalosa) [reply packages/wrangler/src/d1/migrations/apply.tsx:32]: Resolving pending followup

**Review by penalosa — CHANGES_REQUESTED** [review]

**Review by penalosa — APPROVED** [review]

**Pull request merged**